reconcile/templating/renderer.py

```diff
@@ -303,27 +299,22 @@ def reconcile(
             for_each_items: list[dict[str, Any]] = [{}]
             if c.for_each and c.for_each.items:
                 for_each_items = c.for_each.items
-            input = TemplateInput(
+            template_input = TemplateInput(
                 collection=c.name,
                 templates=c.templates,
                 enable_auto_approval=c.enable_auto_approval or False,
                 labels=c.additional_mr_labels or [],
             )
             with PersistenceTransaction(persistence, dry_run) as p:
-                outputs: list[TemplateOutput] = []
                 for item in for_each_items:
-                    outputs.extend(
-                        self.reconcile_template_collection(
-                            c,
-                            gql_no_validation,
-                            p,
-                            ruamel_instance,
-                            input,
-                            item,
-                        )
+                    self.reconcile_template_collection(
+                        collection=c,
+                        gql_api=gql_no_validation,
+                        persistence=p,
+                        ruamel_instance=ruamel_instance,
+                        input=template_input,
+                        each=item,
```

Comment:
output is still needed, see pr https://github.com/app-sre/qontract-reconcile/pull/4488